includes/class-wc-swedbank-plugin.php

```diff
@@ -0,0 +1,488 @@
+					$message = $e->getMessage();
+					WC_Admin_Meta_Boxes::add_error( $message );
+
+					// Rollback
+					$order->update_status(
+						$from,
+						/* translators: 1: note */ sprintf( __( 'Order status rollback. %1$s', 'swedbank-pay-woocommerce-payments' ), $message )
+					);
+				}
+				break;
+			default:
+				// no break
+		}
+
+		// Enable status change hook
+		add_action(
+			'woocommerce_order_status_changed',
+			'\SwedbankPay\Payments\WooCommerce\WC_Swedbank_Plugin::order_status_changed',
+			10
+		);
+	}
+
+	/**
+	 * Add meta boxes in admin
+	 * @return void
+	 */
+	public static function add_meta_boxes() {
+		global $post_id;
+
+		$order = wc_get_order( $post_id );
+
+		if ( $order ) {
+			$payment_method = $order->get_payment_method();
+			if ( in_array( $payment_method, self::PAYMENT_METHODS, true ) ) {
+				$payment_id = get_post_meta( $post_id, '_payex_payment_id', true );
+				if ( ! empty( $payment_id ) ) {
+					add_meta_box(
+						'swedbank_payment_actions',
+						__( 'Swedbank Pay Payments Actions', 'swedbank-pay-woocommerce-payments' ),
+						__CLASS__ . '::order_meta_box_payment_actions',
+						'shop_order',
+						'side',
+						'default'
+					);
+				}
+			}
+		}
+	}
+
+	/**
+	 * MetaBox for Payment Actions
+	 * @return void
+	 */
+	public static function order_meta_box_payment_actions() {
+		global $post_id;
+
+		$order      = wc_get_order( $post_id );
+		$payment_id = get_post_meta( $post_id, '_payex_payment_id', true );
+		if ( empty( $payment_id ) ) {
+			return;
+		}
+
+		// Get Payment Gateway
+		$gateways = WC()->payment_gateways()->get_available_payment_gateways();
+		if ( ! isset( $gateways[ $order->get_payment_method() ] ) ) {
+			return;
+		}
+
+		/** @var \WC_Gateway_Swedbank_Pay_Cc $gateway */
+		$gateway = $gateways[ $order->get_payment_method() ];
+
+		// Fetch payment info
+		try {
+			$result = $gateway->core->fetchPaymentInfo( $payment_id );
+		} catch ( \Exception $e ) {
+			// Request failed
+			return;
+		}
+
+		wc_get_template(
+			'admin/payment-actions.php',
+			array(
+				'gateway'    => $gateway,
+				'order'      => $order,
+				'order_id'   => $post_id,
+				'payment_id' => $payment_id,
+				'info'       => $result,
+			),
+			'',
+			dirname( __FILE__ ) . '/../templates/'
+		);
+	}
+
+	/**
+	 * Add action buttons to Order view
+	 *
+	 * @param WC_Order $order
+	 */
+	public static function add_action_buttons( $order ) {
+		// Get Payment Gateway
+		$payment_method = $order->get_payment_method();
+		if ( in_array( $payment_method, self::PAYMENT_METHODS, true ) ) {
+			$gateways = WC()->payment_gateways()->get_available_payment_gateways();
+
+			if ( isset( $gateways[ $payment_method ] ) ) {
+				/** @var \WC_Gateway_Swedbank_Pay_Cc $gateway */
+				$gateway = $gateways[ $payment_method ];
+
+				wc_get_template(
+					'admin/action-buttons.php',
+					array(
+						'gateway' => $gateway,
+						'order'   => $order,
+					),
+					'',
+					dirname( __FILE__ ) . '/../templates/'
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function dirname() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-checkout/pullRequest?prid=5510424)